### PR TITLE
Add tagger to global registry

### PR DESCRIPTION
**Pull request by pavolloffay**
Updates #107 

Signed-off-by: Pavol Loffay <ploffay@redhat.com>

### CODE REVIEW

File: api/src/main/java/io/opentelemetry/spi/TaggerProvider.java [hunk 1]
```diff
@@ -14,25 +14,23 @@
...
-
-  private Tags() {}
+public interface TaggerProvider {
 
   /**
-   * Returns the default {@code Tagger}.
+   * Creates a new tagger instance.
    *
-   * @return the default {@code Tagger}.
-   * @since 0.1.0
```
Comment (bogdandrutu) [inline api/src/main/java/io/opentelemetry/spi/TaggerProvider.java]: We should probably keep the since tag.

Comment (pavolloffay) [reply api/src/main/java/io/opentelemetry/spi/TaggerProvider.java]: Git recognized this as rename, but it is not a rename.

`Tags.java` has been completely removed

Comment (bogdandrutu) [reply api/src/main/java/io/opentelemetry/spi/TaggerProvider.java]: I meant to have a since tag for every public method.

Comment (pavolloffay) [reply api/src/main/java/io/opentelemetry/spi/TaggerProvider.java]: I will add it in the following PR as it needs to be added in a couple of places in SPI and global registry.

**Comment by pavolloffay** [comment]
I will merge soon, this is just a follow-up PR after moving tracer and metrics to a global registry.

**Pull request merged**